### PR TITLE
[wip]ft-integration of zomato api & redux 

### PR DESCRIPTION
**Pull request by berthoogsteyns**
#### What does this PR do?
* integrate the Zomato api
* state management via Redux
#### Description of Task to be completed?
* refactor restaurantDetail/ - Body to work with redux 
* various bug fixes
#### How should this be manually tested?
* search for 'new york' in the search bar
#### Any background context you want to provide?
Used the redux/toolkit library for generating the reducer & actions
#### Questions:

### CODE REVIEW

File: src/redux/reducers/restaurant.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    },
+    didSearch: (state, action: PayloadAction<number>) => {
+      state.results_fount = action.payload
+    }
+  }
+})
+
+const api = new ApiClient()
+
+export const searchRestaurant = (query: string): AppThunk => async (
```
Comment (kimenyikevin) [inline src/redux/reducers/restaurant.ts]: This function should be in Action

File: src/redux/reducers/restaurant.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    const restaurants = parseToRestaurant(response)
+    dispatch(restaurantSlice.actions.didSearch(response.results_found))
+    dispatch(restaurantSlice.actions.listSuccess(restaurants))
+  } catch (err) {
+    console.log('not found', err)
+    dispatch(restaurantSlice.actions.listFail([]))
+  }
+}
+
+export const setDetail = (restaurant: Restaurant): AppThunk => async (dispatch: AppDispatch) => {
```
Comment (kimenyikevin) [inline src/redux/reducers/restaurant.ts]: And also this one

File: src/redux/reducers/restaurant.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+const initialState: RestaurantState = {
+  list: [],
+  isLoadingList: true,
+  detail: <Restaurant>{},
+  isLoadingDetail: true,
+  results_fount: 0
+}
+
+const restaurantSlice = createSlice({
```
Comment (kimenyikevin) [inline src/redux/reducers/restaurant.ts]: I think this function is the only one which has to be in reducers

File: src/services/api.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      user: parseToUser(r.review.user)
+    }
+    return review
+  })
+  return reviews
+}
+
+export default class ApiClient {
+  private static baseUrl = 'https://developers.zomato.com/api/v2.1'
+  private static apiKey = '47d6af7d73d02359d70879689f24b247'
```
Comment (kimenyikevin) [inline src/services/api.ts]: Add Apikey and BaseUrl to .env file

**Review by kimenyikevin — COMMENTED** [review]
We should have a standard structure for your redux, Like action, reducer, actionType and store.
Make a research on how you can improve the redux structure. 

Let me know when you have Questions.

File: src/redux/reducers/restaurant.ts (hunk 1 above)
Comment (berthoogsteyns) [reply src/redux/reducers/restaurant.ts]: moved actioncreators to a separate file

**Pull request merged**